src/bosh-softlayer-cpi/action/create_vm.go

```diff
@@ -95,7 +95,11 @@ func (a CreateVMAction) updateCloudProperties(cloudProps *VMCloudProperties) {
 		a.vmCloudProperties.Domain = "softlayer.com"
 	}
 	helper.LengthOfHostName = len(a.vmCloudProperties.VmNamePrefix + "." + a.vmCloudProperties.Domain)
-
+	// A workaround for the issue #129 in bosh-softlayer-cpi
+	if helper.LengthOfHostName == 64 {
+		a.vmCloudProperties.VmNamePrefix = a.vmCloudProperties.VmNamePrefix + "-1"
```

Comment:
I don't understand this statement? Why are you appending `"-1"` to the `VmNamePrefix` ? Is it to add two more characters to the `VmNamePrefix`.

Maybe I am missing something simple here but this reads odd... IMO.